cinder/README.md

```diff
@@ -0,0 +1,209 @@
+
+The Dell EMC Cinder container image contains following RPM packages:
+
+- python-storops
+- python-persist-queue
+- python-cachez
+- python-retryz
+- EMC Naviseccli
+
+## VNX deployment
+
+### Prerequisites
+
+- RedHat OpenStack Platform 13.
+- VNX with Block version 5.32 or above.
+
+### Steps
+
+#### Prepare Dell EMC container
+
+The formal Dell EMC container image is published to [Red Hat Container Catalog](https://access.redhat.com/containers/)
+
+> in below examples, 192.168.139.1:8787 acts as a local registry.
+
+Frist, pull the container image from Red Hat Container Catalog.
+
+```bash
+$ docker pull registry.connect.redhat.com/dellemc/openstack-cinder-volume-dellemc
+```
+
+Then, tag and push it to the local docker registry.
+
+```bash
+$ docker tag registry.connect.redhat.com/dellemc/openstack-cinder-volume-dellemc 192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc
+$ docker push 192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc
+```
+
+#### Prepare custom environment yaml
+
+##### Define the custom docker registry
+
+Create or edit `/home/stack/templates/custom-dellemc-container.yaml`.
+
+```yaml
+parameter_defaults:
+  DockerCinderVolumeImage: 192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc
+  DockerInsecureRegistryAddress:
+  - 192.168.139.1:8787
+```
+
+Above adds the director local registry IP `192.168.139.1:8787` to the `undercloud`.
+
+##### Prepare environment yaml for backend
+
+Copy VNX configuration template `cinder-dellemc-vnx-config.yaml` from `tripleo-heat-templates`.
+
+```bash
+cp /usr/share/openstack-tripleo-heat-templates/environments/cinder-dellemc-vnx-config.yaml /home/stack/templates/
+```
+
+Modify backend configuration in `cinder-dellemc-vnx-config.yaml`.
+
+```yaml
+# A Heat environment file which can be used to enable a
+# Cinder Dell EMC VNX backend, configured via puppet
+resource_registry:
+  OS::TripleO::Services::CinderBackendDellEMCVNX: /usr/share/openstack-tripleo-heat-templates/puppet/services/cinder-backend-dellemc-vnx.yaml
+
+parameter_defaults:
+  CinderEnableIscsiBackend: false
+  CinderEnableRbdBackend: false
+  CinderEnableNfsBackend: false
+  NovaEnableRbdBackend: false
+  GlanceBackend: file
+  CinderEnableDellEMCVNXBackend: true
+  CinderDellEMCVNXBackendName: 'tripleo_dellemc_vnx'
+  CinderDellEMCVNXSanIp: '192.168.1.50'
+  CinderDellEMCVNXSanLogin: 'admin'
+  CinderDellEMCVNXSanPassword: 'password'
+  CinderDellEMCVNXStorageProtocol: 'iscsi'
+  CinderDellEMCVNXStoragePoolName: ''
+  CinderDellEMCVNXDefaultTimeout: 3600
+  CinderDellEMCVNXMaxLunsPerStorageGroup: 255
+  CinderDellEMCVNXInitiatorAutoRegistration: 'true'
+  CinderDellEMCVNXAuthType: 'global'
+  CinderDellEMCVNXStorageSecurityFileDir: ''
+  CinderDellEMCVNXNaviSecCliPath: '/opt/Navisphere/bin/naviseccli'
+```
+
+For a full detailed instruction of options, please refer to [VNX backend configuration](https://docs.openstack.org/cinder/queens/configuration/block-storage/drivers/dell-emc-vnx-driver.html)
```

Comment:
Shouldn't we provide the latest doc: https://docs.openstack.org/cinder/latest/configuration/block-storage/drivers/dell-emc-vnx-driver.html

Reply:
Done